demos/coreMQTT/mqtt_demo_keep_alive.c

```diff
@@ -0,0 +1,1177 @@
+
+        if( xResult != MQTTSuccess )
+        {
+            LogError( ( "MQTT_Subscribe() failed with status %s.",
+                        MQTT_Status_strerror( xResult ) ) );
+            xReturnStatus = pdFAIL;
+            break;
+        }
+
+        LogInfo( ( "SUBSCRIBE sent for topic %s to broker.\n\n", mqttexampleTOPIC ) );
+
+        /* Process incoming packet from the broker. After sending the subscribe, the
+         * client may receive a publish before it receives a subscribe ack. Therefore,
+         * call generic incoming packet processing function. Since this demo is
+         * subscribing to the topic to which no one is publishing, probability of
+         * receiving publish message before subscribe ack is zero; but application
+         * must be ready to receive any packet.  This demo uses the generic packet
+         * processing function everywhere to highlight this fact. */
+        xResult = MQTT_ProcessLoop( pxMQTTContext, mqttexampleRECEIVE_LOOP_TIMEOUT_MS );
+
+        if( xResult != MQTTSuccess )
+        {
+            LogError( ( "MQTT_ProcessLoop() failed with status %s.",
+                        MQTT_Status_strerror( xResult ) ) );
+            xReturnStatus = pdFAIL;
+            break;
+        }
+
+        /* Check if recent subscription request has been rejected. #xTopicFilterContext is updated
+         * in the event callback to reflect the status of the SUBACK sent by the broker. It represents
+         * either the QoS level granted by the server upon subscription, or acknowledgement of
+         * server rejection of the subscription request. */
+        for( ulTopicCount = 0; ulTopicCount < mqttexampleTOPIC_COUNT; ulTopicCount++ )
+        {
+            if( xTopicFilterContext[ ulTopicCount ].xSubAckStatus == MQTTSubAckFailure )
+            {
+                LogWarn( ( "Server rejected subscription request. Attempting to re-subscribe to topic %s.",
+                           xTopicFilterContext[ ulTopicCount ].pcTopicFilter ) );
+                xFailedSubscribeToTopic = true;
+                xRetryUtilsStatus = RetryUtils_BackoffAndSleep( &xRetryParams );
+                break;
+            }
+        }
+
+        if( xRetryUtilsStatus == RetryUtilsRetriesExhausted )
+        {
+            LogError( ( "Failed to subscribe to the broker: Subscription request"
+                        " retires exhausted. NumberOfAttempts=%ul",
+                        ( unsigned long ) ( xRetryParams.maxRetryAttempts ) ) );
+            xReturnStatus = pdFAIL;
+        }
+    } while( ( xFailedSubscribeToTopic == true ) && ( xRetryUtilsStatus == RetryUtilsSuccess ) );
+
+    return xReturnStatus;
+}
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvMQTTPublishToTopic( MQTTContext_t * pxMQTTContext )
+{
+    MQTTStatus_t xResult;
+    MQTTPublishInfo_t xMQTTPublishInfo;
+    BaseType_t xTimerStatus;
+    BaseType_t xReturnStatus = pdPASS;
```

Comment:
Here, the `xTimerStatus` can be avoided as well.
```suggestion
    BaseType_t xReturnStatus = pdPASS;
```